Compilation from source on macOS fails in libuv

Running `make USE_BINARYBUILDER=0 VERBOSE=1` on macOS 12 on julia master gives:

```
/private/tmp/julia/deps/srccache/libuv-f872be3b6438b90b4690fe3ee7692c50bfd9c7c7/src/unix/process.c:290:12: error: implicit declaration of function 'execvpe' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    return execvpe(file, argv, envp);
           ^
1 error generated.
make[2]: *** [src/unix/libuv_la-process.lo] Error 1
make[1]: *** [scratch/libuv-f872be3b6438b90b4690fe3ee7692c50bfd9c7c7/build-compiled] Error 2
make: *** [julia-deps] Error 2
```

The code in question is:

```
#ifdef __GLIBC_PREREQ
# if (__GLIBC_PREREQ(2,11))
  /* Old glibc did not have execvpe at all, so we always emulate it. */
  if (envp == environ)
    return execvpe(file, argv, envp);
#endif
#else
  /* Assume all other environments have execvpe. */
  if (envp == environ)
    return execvpe(file, argv, envp);
#endif
```

but `execvpe` is not available on macOS. I suppose either the code should be fixed, or be compiled with `-Wno-implicit-function-declaration` on compilers that enable it by default (clang).

--- a/src/unix/process.c
+++ b/src/unix/process.c
@@ -257,7 +257,7 @@ static const char* uv__spawn_find_path_in_env(char* const* env) {
   return NULL;
 }
 
-
+#ifdef __linux__
 static int uv__execvpe(const char *file, char *const argv[], char *const envp[]) {
   const char *p;
   const char *z;
@@ -352,6 +352,7 @@ static int uv__execvpe(const char *file, char *const argv[], char *const envp[])
     return EACCES;
   return err;
 }
+#endif
 
 
 static void uv__write_int(

